File: samples/src/main/java/jvmMain/java/BlazeSample.java
package jvmMain.java;

import static jvmMain.java.Config.*;
import static one.mixin.bot.blaze.BlazeClientKt.sendTextMsg;

import java.io.IOException;

import okhttp3.WebSocket;
import one.mixin.bot.blaze.BlazeClient;
import one.mixin.bot.blaze.BlazeHandler;
import one.mixin.bot.blaze.BlazeMsg;
import one.mixin.bot.blaze.MsgData;
import one.mixin.bot.extension.ByteArrayExtensionKt;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("SameParameterValue")
class BlazeSample {
  public static void main(String[] args) throws IOException {

    BlazeClient blazeClient =
        new BlazeClient.Builder()
            .configSafeUser(BOT_USER_ID, BOT_SESSION_ID, ByteArrayExtensionKt.hexStringToByteArray(BOT_SESSION_PRIVATE_KEY), null, null)
            .enableDebug()
            .enableParseData()
            .enableAutoAck()
            .blazeHandler(new MyBlazeHandler())
            .build();
    blazeClient.start();

    // 卡一下；因为没有任何别的服务启动
    //noinspection ResultOfMethodCallIgnored
    System.in.read();
  }

  private static class MyBlazeHandler implements BlazeHandler {

    @Override
    public boolean onMessage(@NotNull WebSocket webSocket, @NotNull BlazeMsg blazeMsg) {
      System.out.println(blazeMsg);
      MsgData data = blazeMsg.getData();
      if (data != null) {
        sendTextMsg(webSocket, data.getConversionId(), data.getUserId(), "read");
      }

      return true;
    }
  }
}

File: samples/src/main/java/jvmMain/kotlin/BlazeSample.kt
package jvmMain.kotlin

import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import okhttp3.WebSocket
import one.mixin.bot.blaze.BlazeClient
import one.mixin.bot.blaze.BlazeHandler
import one.mixin.bot.blaze.BlazeMsg
import one.mixin.bot.blaze.sendTextMsg
import one.mixin.bot.extension.base64Decode
import one.mixin.bot.extension.hexStringToByteArray
import one.mixin.bot.util.newKeyPairFromPrivateKey
import one.mixin.bot.util.newKeyPairFromSeed

fun main(): Unit = runBlocking {
    val job = launch {
        val keyPair = newKeyPairFromSeed(Config.BOT_SESSION_PRIVATE_KEY.hexStringToByteArray())
        val blazeClient =
            BlazeClient.Builder().configSafeUser(Config.BOT_USER_ID, Config.BOT_SESSION_ID, keyPair.privateKey)
                .enableDebug().enableParseData().enableAutoAck().blazeHandler(MyBlazeHandler()).build()
        blazeClient.start()
    }
    job.join()
}

private class MyBlazeHandler : BlazeHandler {
    override fun onMessage(
        webSocket: WebSocket,
        blazeMsg: BlazeMsg,
    ): Boolean {
        println(blazeMsg)
        blazeMsg.data?.let { data ->
            sendTextMsg(webSocket, data.conversionId, data.userId, "read")
        }
        return true
    }
}
